### ISSUE TITLE
hatch_path panics when used as shown in documentation

### ISSUE THREAD

**Issue by concave-sphere** [nical/lyon#634]
There may be something I'm missing, but the following code panics for me when `hatch_path` is invoked ([full code archive](https://github.com/nical/lyon/files/5973751/lyon_hatch.tar.gz)
), using both lyon_algorithms 0.17.1 and git.  As far as I can tell, what I'm doing should behave identically to the example in the [module documentation](https://docs.rs/lyon_algorithms/0.17.1/lyon_algorithms/hatching/index.html).

```
    fn transform(&self, path: &lyon_path::Path) -> lyon_path::Path {
        // ... setup elided ...

        let mut hatched_path_builder = Path::builder();
        let mut hatcher = Hatcher::new();
        hatcher.hatch_path(
            path.iter(),
            &hatching_options,
            &mut RegularHatchingPattern {
                interval: self.spacing,
                callback: &mut |segment: &HatchSegment| {
                    hatched_path_builder.add_line_segment(&LineSegment {
                        from: segment.a.position,
                        to: segment.b.position,
                    });
                },
            },
        );

        hatched_path_builder.build()
    }
```

The backtrace looks like this:

```
---- tests::straight_lines_hatch stdout ----
thread 'tests::straight_lines_hatch' panicked at 'assertion failed: self.in_subpath', /home/daniel/.cargo/git/checkouts/lyon-985c156d9631db61/74e6b13/path/src/private.rs:36:13
stack backtrace:
   0: rust_begin_unwind
             at /rustc/cb75ad5db02783e8b0222fee363c5f63f7e2cf5b/library/std/src/panicking.rs:493:5
   1: core::panicking::panic_fmt
             at /rustc/cb75ad5db02783e8b0222fee363c5f63f7e2cf5b/library/core/src/panicking.rs:92:14
   2: core::panicking::panic
             at /rustc/cb75ad5db02783e8b0222fee363c5f63f7e2cf5b/library/core/src/panicking.rs:50:5
   3: lyon_path::private::DebugValidator::end
             at /home/daniel/.cargo/git/checkouts/lyon-985c156d9631db61/74e6b13/path/src/private.rs:36:13
   4: <lyon_algorithms::hatching::EventsBuilder as lyon_path::builder::PathBuilder>::end
             at /home/daniel/.cargo/git/checkouts/lyon-985c156d9631db61/74e6b13/algorithms/src/hatching.rs:521:9
   5: lyon_path::builder::PathBuilder::close
             at /home/daniel/.cargo/git/checkouts/lyon-985c156d9631db61/74e6b13/path/src/builder.rs:143:9
   6: <lyon_algorithms::hatching::EventsBuilder as lyon_path::builder::Build>::build
             at /home/daniel/.cargo/git/checkouts/lyon-985c156d9631db61/74e6b13/algorithms/src/hatching.rs:498:9
   7: lyon_algorithms::hatching::HatchingEvents::set_path
             at /home/daniel/.cargo/git/checkouts/lyon-985c156d9631db61/74e6b13/algorithms/src/hatching.rs:554:30
   8: lyon_algorithms::hatching::Hatcher::hatch_path
             at /home/daniel/.cargo/git/checkouts/lyon-985c156d9631db61/74e6b13/algorithms/src/hatching.rs:316:9
   9: <lyon_hatch::StraightLinesHatch as lyon_hatch::PathTransformer>::transform
             at ./src/lib.rs:23:9
  10: lyon_hatch::tests::straight_lines_hatch
             at ./src/lib.rs:59:9
  11: lyon_hatch::tests::straight_lines_hatch::{{closure}}
             at ./src/lib.rs:46:5
  12: core::ops::function::FnOnce::call_once
             at /rustc/cb75ad5db02783e8b0222fee363c5f63f7e2cf5b/library/core/src/ops/function.rs:227:5
  13: core::ops::function::FnOnce::call_once
             at /rustc/cb75ad5db02783e8b0222fee363c5f63f7e2cf5b/library/core/src/ops/function.rs:227:5
```

**Comment by nical** [comment]
Thanks for filing this and providing a test case. I'll  have a look soon.

**Comment by nical** [comment]
Sorry for the delay. The fix is available in lyon_algorithms 0.17.2, it should work once you refresh your dependencies.

**Comment by concave-sphere** [comment]
Yep, works now.  Thanks!
